tests/tree_inference/test_mcmc.py

```diff
@@ -191,3 +193,161 @@ def test_swap_subtrees_move():
+                [0, 0, 0, 0, 0, 1, 0, 0, 0],  # 5
+                [0, 0, 0, 0, 0, 0, 0, 0, 0],  # 6
+                [0, 0, 0, 0, 0, 0, 0, 0, 0],  # 7
+                [0, 0, 0, 0, 0, 0, 0, 0, 0],  # 8
+                [1, 0, 1, 0, 1, 0, 0, 0, 0],  # R
+            ]
+        ),
+        labels=jnp.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),
+    )
+    # if the i node was attached node 2
+    new_tree_corr2 = Tree(
+        jnp.array(
+            [  # 1, 2, 3, 4, 5, 6, 7, 8, R
+                [0, 1, 0, 1, 0, 0, 1, 0, 0],  # 1
+                [0, 0, 0, 0, 0, 0, 0, 1, 0],  # 2
+                [0, 0, 0, 0, 0, 0, 0, 0, 0],  # 3
+                [0, 0, 0, 0, 0, 0, 0, 0, 0],  # 4
+                [0, 0, 0, 0, 0, 1, 0, 0, 0],  # 5
+                [0, 0, 0, 0, 0, 0, 0, 0, 0],  # 6
+                [0, 0, 0, 0, 0, 0, 0, 0, 0],  # 7
+                [0, 0, 0, 0, 0, 0, 0, 0, 0],  # 8
+                [1, 0, 1, 0, 1, 0, 0, 0, 0],  # R
+            ]
+        ),
+        labels=jnp.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),
+    )
+    # if the i node was attached node 4
+    new_tree_corr3 = Tree(
+        jnp.array(
+            [  # 1, 2, 3, 4, 5, 6, 7, 8, R
+                [0, 1, 0, 1, 0, 0, 1, 0, 0],  # 1
+                [0, 0, 0, 0, 0, 0, 0, 0, 0],  # 2
+                [0, 0, 0, 0, 0, 0, 0, 0, 0],  # 3
+                [0, 0, 0, 0, 0, 0, 0, 1, 0],  # 4
+                [0, 0, 0, 0, 0, 1, 0, 0, 0],  # 5
+                [0, 0, 0, 0, 0, 0, 0, 0, 0],  # 6
+                [0, 0, 0, 0, 0, 0, 0, 0, 0],  # 7
+                [0, 0, 0, 0, 0, 0, 0, 0, 0],  # 8
+                [1, 0, 1, 0, 1, 0, 0, 0, 0],  # R
+            ]
+        ),
+        labels=jnp.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),
+    )
+    # if the i node was attached node 7
+    new_tree_corr4 = Tree(
+        jnp.array(
+            [  # 1, 2, 3, 4, 5, 6, 7, 8, R
+                [0, 1, 0, 1, 0, 0, 1, 0, 0],  # 1
+                [0, 0, 0, 0, 0, 0, 0, 0, 0],  # 2
+                [0, 0, 0, 0, 0, 0, 0, 0, 0],  # 3
+                [0, 0, 0, 0, 0, 0, 0, 0, 0],  # 4
+                [0, 0, 0, 0, 0, 1, 0, 0, 0],  # 5
+                [0, 0, 0, 0, 0, 0, 0, 0, 0],  # 6
+                [0, 0, 0, 0, 0, 0, 0, 1, 0],  # 7
+                [0, 0, 0, 0, 0, 0, 0, 0, 0],  # 8
+                [1, 0, 1, 0, 1, 0, 0, 0, 0],  # R
+            ]
+        ),
+        labels=jnp.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),
+    )
+
+    # check if any of the four possible trees is the same as the new tree
+    assert (
+        jnp.array_equal(new_tree.tree_topology, new_tree_corr1.tree_topology)
+        or jnp.array_equal(new_tree.tree_topology, new_tree_corr2.tree_topology)
+        or jnp.array_equal(new_tree.tree_topology, new_tree_corr3.tree_topology)
+        or jnp.array_equal(new_tree.tree_topology, new_tree_corr4.tree_topology)
+    )
+    # check if the labels are the same, just once for sanity
+    assert jnp.array_equal(new_tree.labels, new_tree_corr1.labels)
```

Comment:
Implements the different lineage subtree swap as in the SCITE supplement. 

![Screenshot 2023-05-30 at 13 02 38](https://github.com/cbg-ethz/PYggdrasil/assets/10867778/b797ea1d-887b-453b-9553-f9116effc69d)

Reply:
Great job! :slightly_smiling_face: